docs/explanation/security.md

```diff
@@ -0,0 +1,70 @@
+
+For example, the permissions of the databases are:
+
+```text
+-rw-r--r--  1 root root  32K Jul 13 23:48 .tracing-data.db
+-rw-------  1 root root  20K Jul 13 23:48 .unit-state.db
+```
+
+## ops[testing]
+
+When testing an event with [](ops.testing.Context), the mocked unit state database and tracing data are stored in memory. Each event creates a new charm directory, which is provided by [](tempfile.TemporaryDirectory).
+
+## Security updates
+
+We strongly recommend restricting the version of `ops` (and `ops[harness,testing,tracing]` in your dev dependencies) in `pyproject.toml` in a way that allows picking up new compatible releases every time that you re-lock. If your charm needs to support Ubuntu 20.04 (with Python 3.8), then this looks like `ops~=2.23`. Otherwise, this looks like `ops~=3.0`. Set a minor version that includes all the features that the charm uses.
+
+Your charm repository should have tooling configured so that any dependencies with security updates are detected automatically (such as [Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-security-updates/about-dependabot-security-updates) or [Renovate](https://www.mend.io/renovate/)), prompting to you re-lock so that the charm will be built with the latest version.
+
+For information about supported versions and how to report security issues, see [SECURITY.md](https://github.com/canonical/operator/blob/main/SECURITY.md).
+
+## Risks
+
+Ops inherits the risks of Juju executing charms (for example, injecting data into the charm context or through the use of hook commands). Charm authors should be familiar with {external+juju:ref}`Juju security <juju-security>` and {external+pebble:ref}` Pebble security <security>`.
+
+If a charm is integrated with a tracing receiver, Ops introduces the risk of outgoing traces being intercepted. Traces should not include any sensitive data, but intercepted traces can provide information about the structure of the charm and the events that the charm has processed. In addition, an attacker that blocked trace data could hide malicious activity.
+
+Otherwise, Ops doesn't introduce any new security risks. Ops does expand the impact of the Juju risk of an attacker gaining access to the filesystem of the charm:
+
+* Access to the deferred notice queue provides information about events that could not be immediately processed (this includes the event name and the event context provided in the hook environment at the time).
+* Access to trace data provides detailed information about the implementation of the charm and the events that it has processed. This is particularly the case when tracing has not been configured, as the charm will have a large amount of buffered trace data stored (when tracing is active, this will be regularly sent to the trace receiver and removed from the local database).
+
+## Good practices
```

Comment:
Depending on the answer to scope, this section may or may not belong here.

It could be dropped or moved to how-to perhaps.

Reply:
The guidance for this page says to include:

> topics such as common risks, good practices, built-in protection and so on

So that's what it has :smile:.

I think some how-to guides would be nice, and the guidance suggests those too, but they aren't required, so I'd rather leave them for future work than include them in this cycle's roadmap item.